docs/template_plugin/tests/functional/op_reference/asinh.cpp

```diff
@@ -0,0 +1,103 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <gtest/gtest.h>
+
+#include <ie_core.hpp>
+#include <ie_ngraph_utils.hpp>
+#include <ngraph/ngraph.hpp>
+#include <shared_test_classes/base/layer_test_utils.hpp>
+#include <vector>
+
+#include "base_reference_test.hpp"
+
+using namespace ngraph;
+
+namespace reference_tests {
+namespace {
+
+struct AsinhParams {
+    Tensor input;
+    Tensor expected;
+};
+
+struct Builder : ParamsBuilder<AsinhParams> {
+    REFERENCE_TESTS_ADD_SET_PARAM(Builder, input);
+    REFERENCE_TESTS_ADD_SET_PARAM(Builder, expected);
+};
+
+class ReferenceAsinhLayerTest : public testing::TestWithParam<AsinhParams>, public CommonReferenceTest {
+public:
+    void SetUp() override {
+        auto params = GetParam();
+        function = CreateFunction(params.input.shape, params.input.type);
+        inputData = {params.input.data};
+        refOutData = {params.expected.data};
+    }
+    static std::string getTestCaseName(const testing::TestParamInfo<AsinhParams>& obj) {
+        auto param = obj.param;
+        std::ostringstream result;
+        result << "shape=" << param.input.shape << "_";
+        result << "type=" << param.input.type;
+        return result.str();
+    }
+
+private:
+    static std::shared_ptr<Function> CreateFunction(const Shape& shape, const element::Type& type) {
+        const auto in = std::make_shared<op::Parameter>(type, shape);
+        const auto Asinh = std::make_shared<op::Asinh>(in);
+        return std::make_shared<Function>(NodeVector {Asinh}, ParameterVector {in});
+    }
+};
+
+TEST_P(ReferenceAsinhLayerTest, AsinhWithHardcodedRefs) {
+    Exec();
+}
+
+}  // namespace
+
+INSTANTIATE_TEST_SUITE_P(
+    smoke_Asinh_With_Hardcoded_Refs, ReferenceAsinhLayerTest,
+    ::testing::Values(
+        Builder {}
+            .input({{11}, element::f16, std::vector<ngraph::float16> {0.f, 1.f, -1.f, 2.f, -2.f, 3.f, -3.f, 4.f, 5.f, 10.f, 100.f}})
```

Comment:
I think, it will better to use such [mechanism](https://github.com/openvinotoolkit/openvino/blob/243028a40871dd9b2ed95f92c75063ebd1b140c1/docs/template_plugin/tests/functional/op_reference/gather_tree.cpp#L66). I mean to use templates to combine different datatypes for vectors

Reply:
Sometimes I need to modify each element of expected values when int ot uint case. I mean that the value from backend test used -2 and 2 for input value -3 and 3 in data type int32/64. But type cast returns -1 and 1 from -1.81844646f and 1.81844646f and they causes test failure. So I think that template type is not proper for this case.